tests/test_dataclasses.py

```diff
@@ -795,3 +799,34 @@ class Config:
     e.other = 'bulbi2'
     with pytest.raises(dataclasses.FrozenInstanceError):
         e.item.name = 'pika2'
+
+
+@dataclasses.dataclass
+class BuiltInDataclassForPickle:
+    value: int
+
+
+class ModelForPickle(pydantic.BaseModel):
+    """
+    pickle can only work with top level classes as it imports them
+    """
+
+    dataclass: BuiltInDataclassForPickle
+
+    class Config:
+        validate_assignment = True
+
+
+def test_pickle_overriden_builtin_dataclass():
+    value = 5
+    obj = ModelForPickle(dataclass=BuiltInDataclassForPickle(value=value))
+
+    pickled_obj = pickle.dumps(obj)
+    restored_obj = pickle.loads(pickled_obj)
+
+    assert restored_obj.dataclass.value == value
+    assert restored_obj == obj
+
+    # ensure the restored dataclass is still a pydantic dataclass
+    with pytest.raises(ValidationError):
```

Comment:
```suggestion
    with pytest.raises(ValidationError, match='...'):
```
Add something that confirms we're catching the error we think we are.